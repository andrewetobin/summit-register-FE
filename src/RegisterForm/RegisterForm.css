@import url("https://fonts.googleapis.com/css?family=Archivo");

* {
  font-family: "Archivo", sans-serif;
}

.RegisterForm {
  align-items: center;
  background: url("../utilities/images/mountains-2.jpg") no-repeat center
    rgba(165, 165, 165, 0.68);
  background-size: auto 70%;
  display: flex;
  flex-direction: column;
  height: 90vh;

  border: 4px solid black;
  border-radius: 10%;
  margin: 0 35%;
  width: 30%;
}

.main-title {
  color: white;
  font-size: 30px;
  margin: 10px 0 0 0;
}

.logo {
  height: 40px;
  margin-left: 10px;
  position: relative;
  top: 8px;
}

.drop-down-menu {
  font-size: 12px;
  padding: 5px 10px;
}

.select-btn {
  border: 1px solid black;
  border-radius: 20px;
  font-size: 12px;
  margin-left: 5px;
  padding: 2px 10px;
}

.drop-down-controls {
  margin-top: 90%;
}

.sign-register-btn {
  border: 1px solid black;
  border-radius: 20px;
  display: block;
  font-size: 16px;
  margin: 70px 0;
  padding: 5px 10px;
}

.select-btn:hover,
.sign-register-btn:hover {
  background: black;
  color: white;
  transition: all 0.4s ease-in-out;
}
