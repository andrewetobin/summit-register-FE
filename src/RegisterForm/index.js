import React, { Component } from "react";
import PropTypes from "prop-types";
import "./RegisterForm.css";

class RegisterForm extends Component {
  constructor() {
    super();
    this.state = {
      name: "",
      hometown: "",
      comments: ""
    };
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = e => {
    e.preventDefault();
    this.props.handleLogUpdate(this.state);
  };

  render() {
    const { name, hometown, comments } = this.state;
    return (
      <div className="RegisterForm">
        <form className="register-form">
          <h3 className="form-instructions">
            Name:
            <input
              className="input"
              value={name}
              name="name"
              onChange={this.handleChange}
            />
          </h3>
          <h3 className="form-instructions">
            Hometown:
            <input
              className="input"
              value={hometown}
              name="hometown"
              onChange={this.handleChange}
            />
          </h3>
          <h3 className="form-instructions">
            Comments:
            <textarea
              className="input"
              value={comments}
              name="comments"
              onChange={this.handleChange}
            />
          </h3>
        </form>
      </div>
    );
  }
}

RegisterForm.propTypes = {
  currentDisplay: PropTypes.string,
  handleBackButton: PropTypes.func,
  handleLogUpdate: PropTypes.func
};
export default RegisterForm;
