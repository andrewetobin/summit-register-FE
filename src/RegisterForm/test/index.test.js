import React from "react";
import ReactDOM from "react-dom";
import RegisterForm from "..";
import { shallow } from "enzyme";

jest.mock("../../utilities/helper/apiCalls");

describe("RegisterForm", () => {
  let wrapper;
  let mockHandleLogUpdate;
  const mockPostImage = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <RegisterForm handleLogUpdate={jest.fn()} mockPostImage={{}} />
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<RegisterForm handleLogUpdate={jest.fn()} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  describe("handleChange", () => {
    it("should update state on name change", () => {
      const event = { target: { name: "name", value: "Chris" } };
      wrapper.instance().handleChange(event);
      expect(wrapper.state().name).toEqual("Chris");
    });

    it("should update state on hometown change", () => {
      const event = { target: { name: "hometown", value: "Denver" } };
      wrapper.instance().handleChange(event);
      expect(wrapper.state().hometown).toEqual("Denver");
    });

    it("should update state on comments change", () => {
      const event = { target: { name: "comments", value: "Hiking rules" } };
      wrapper.instance().handleChange(event);
      expect(wrapper.state().comments).toEqual("Hiking rules");
    });
  });

  describe("handleSubmit", () => {
    it("should call handleSubmit on submit button click", () => {
      const spy = spyOn(wrapper.instance(), "handleSubmit");
      const mockEvent = { preventDefault: jest.fn() };
      wrapper.instance().forceUpdate();

      wrapper.find(".submit-btn").simulate("click", mockEvent);
      expect(spy).toHaveBeenCalled();
    });
  });

  describe("handleImage", () => {
    it("should set state of image url", () => {
      let expected = "something";
      let mockEventFile = {
        target: {
          files: ["something"]
        }
      };

      wrapper.instance().handleImage(mockEventFile);
      expect(wrapper.state().image).toEqual("something");
    });
  });
});
