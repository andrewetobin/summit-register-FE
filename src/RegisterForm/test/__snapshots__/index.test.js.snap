// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegisterForm
    handleLogUpdate={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="register-form"
        >
          <h3
            className="form-instructions"
          >
            Name:
            <input
              className="input"
              name="name"
              onChange={[Function]}
              value=""
            />
          </h3>
          <h3
            className="form-instructions"
          >
            Hometown:
            <input
              className="input"
              name="hometown"
              onChange={[Function]}
              value=""
            />
          </h3>
          <h3
            className="form-instructions"
          >
            Comments:
            <textarea
              className="input"
              name="comments"
              onChange={[Function]}
              value=""
            />
          </h3>
          <input
            accept="image/*"
            onChange={[Function]}
            type="file"
          />
        </form>,
        <button
          className="submit-btn"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          Submit
        </button>,
      ],
      "className": "RegisterForm",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="form-instructions"
            >
              Name:
              <input
                className="input"
                name="name"
                onChange={[Function]}
                value=""
              />
            </h3>,
            <h3
              className="form-instructions"
            >
              Hometown:
              <input
                className="input"
                name="hometown"
                onChange={[Function]}
                value=""
              />
            </h3>,
            <h3
              className="form-instructions"
            >
              Comments:
              <textarea
                className="input"
                name="comments"
                onChange={[Function]}
                value=""
              />
            </h3>,
            <input
              accept="image/*"
              onChange={[Function]}
              type="file"
            />,
          ],
          "className": "register-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Name:",
                <input
                  className="input"
                  name="name"
                  onChange={[Function]}
                  value=""
                />,
              ],
              "className": "form-instructions",
            },
            "ref": null,
            "rendered": Array [
              "Name:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input",
                  "name": "name",
                  "onChange": [Function],
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Hometown:",
                <input
                  className="input"
                  name="hometown"
                  onChange={[Function]}
                  value=""
                />,
              ],
              "className": "form-instructions",
            },
            "ref": null,
            "rendered": Array [
              "Hometown:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input",
                  "name": "hometown",
                  "onChange": [Function],
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Comments:",
                <textarea
                  className="input"
                  name="comments"
                  onChange={[Function]}
                  value=""
                />,
              ],
              "className": "form-instructions",
            },
            "ref": null,
            "rendered": Array [
              "Comments:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input",
                  "name": "comments",
                  "onChange": [Function],
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "accept": "image/*",
              "onChange": [Function],
              "type": "file",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "submit-btn",
          "disabled": true,
          "onClick": [Function],
          "type": "submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="register-form"
          >
            <h3
              className="form-instructions"
            >
              Name:
              <input
                className="input"
                name="name"
                onChange={[Function]}
                value=""
              />
            </h3>
            <h3
              className="form-instructions"
            >
              Hometown:
              <input
                className="input"
                name="hometown"
                onChange={[Function]}
                value=""
              />
            </h3>
            <h3
              className="form-instructions"
            >
              Comments:
              <textarea
                className="input"
                name="comments"
                onChange={[Function]}
                value=""
              />
            </h3>
            <input
              accept="image/*"
              onChange={[Function]}
              type="file"
            />
          </form>,
          <button
            className="submit-btn"
            disabled={true}
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>,
        ],
        "className": "RegisterForm",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="form-instructions"
              >
                Name:
                <input
                  className="input"
                  name="name"
                  onChange={[Function]}
                  value=""
                />
              </h3>,
              <h3
                className="form-instructions"
              >
                Hometown:
                <input
                  className="input"
                  name="hometown"
                  onChange={[Function]}
                  value=""
                />
              </h3>,
              <h3
                className="form-instructions"
              >
                Comments:
                <textarea
                  className="input"
                  name="comments"
                  onChange={[Function]}
                  value=""
                />
              </h3>,
              <input
                accept="image/*"
                onChange={[Function]}
                type="file"
              />,
            ],
            "className": "register-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Name:",
                  <input
                    className="input"
                    name="name"
                    onChange={[Function]}
                    value=""
                  />,
                ],
                "className": "form-instructions",
              },
              "ref": null,
              "rendered": Array [
                "Name:",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "input",
                    "name": "name",
                    "onChange": [Function],
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Hometown:",
                  <input
                    className="input"
                    name="hometown"
                    onChange={[Function]}
                    value=""
                  />,
                ],
                "className": "form-instructions",
              },
              "ref": null,
              "rendered": Array [
                "Hometown:",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "input",
                    "name": "hometown",
                    "onChange": [Function],
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Comments:",
                  <textarea
                    className="input"
                    name="comments"
                    onChange={[Function]}
                    value=""
                  />,
                ],
                "className": "form-instructions",
              },
              "ref": null,
              "rendered": Array [
                "Comments:",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "input",
                    "name": "comments",
                    "onChange": [Function],
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "accept": "image/*",
                "onChange": [Function],
                "type": "file",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "submit-btn",
            "disabled": true,
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
